third_party/bigframes_vendored/sklearn/decomposition/_mf.py

```diff
@@ -0,0 +1,93 @@
+""" Matrix Factorization.
+"""
+
+# Author: Alexandre Gramfort <alexandre.gramfort@inria.fr>
+#         Olivier Grisel <olivier.grisel@ensta.org>
+#         Mathieu Blondel <mathieu@mblondel.org>
+#         Denis A. Engemann <denis-alexander.engemann@inria.fr>
+#         Michael Eickenberg <michael.eickenberg@inria.fr>
+#         Giorgio Patrini <giorgio.patrini@anu.edu.au>
+#
+# License: BSD 3 clause
+
+from abc import ABCMeta
+
+from bigframes_vendored.sklearn.base import BaseEstimator
+
+from bigframes import constants
+
+
+class MatrixFactorization(BaseEstimator, metaclass=ABCMeta):
+    """Matrix Factorization (MF).
+
+    **Examples:**
+
+        >>> import bigframes.pandas as bpd
+        >>> from bigframes.ml.decomposition import MatrixFactorization
+        >>> X = bpd.DataFrame([[1, 1], [2, 1], [3, 1.2], [4, 1], [5, 0.8], [6, 1]])
+        >>> model = MatrixFactorization(n_components=2, init='random', random_state=0)
+        >>> W = model.fit_transform(X)
+        >>> H = model.components_
+
+    Args:
+        num_factors (int or auto, default auto):
+            Specifies the number of latent factors to use.
+            If you aren't running hyperparameter tuning, then you can specify an INT64 value between 2 and 200. The default value is log2(n), where n is the number of training examples.
+        user_col (str):
+            The user column name.
+        item_col (str):
+            The item column name.
+        l2_reg (float, default 1.0):
+            If you aren't running hyperparameter tuning, then you can specify a FLOAT64 value. The default value is 1.0.
+            If you are running hyperparameter tuning, then you can use one of the following options:
+                The HPARAM_RANGE keyword and two FLOAT64 values that define the range to use for the hyperparameter. For example, L2_REG = HPARAM_RANGE(1.5, 5.0).
+                The HPARAM_CANDIDATES keyword and an array of FLOAT64 values that provide discrete values to use for the hyperparameter. For example, L2_REG = HPARAM_CANDIDATES([0, 1.0, 3.0, 5.0]).
```

Comment:
The command `nox -r -s docs` is failing. I believe it's because of these lines.

```
sphinx.errors.SphinxWarning: /home/runner/work/python-bigquery-dataframes/python-bigquery-dataframes/bigframes/ml/decomposition.py:docstring of bigframes.ml.decomposition.MatrixFactorization:21:Unexpected indentation.
```

Since we aren't supporting hyperparameter tuning yet (internal issue 384510566), please remove references to these features.

```suggestion
```